test/commands/generate/index.test.ts

```diff
@@ -1,19 +1,24 @@
-import { test } from '@oclif/test';
+import { test, expect } from '@oclif/test';
+import * as fs from 'fs-extra';
 import * as inquirer from 'inquirer';
-import * as sinon from 'sinon';
 
 describe('running command generate', () => {
-  const stubResponse = sinon.stub()
-    .onFirstCall()
-    .resolves({ platform: 'aws' })
-    .onSecondCall()
-    .resolves({ infrastructureType: 'advance' });
+  const projectName = 'app-name';
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    fs.removeSync(projectName);
+  });
+
+  const stubResponse = jest.fn();
+  stubResponse.mockReturnValueOnce({ platform: 'aws' });
+  stubResponse.mockReturnValueOnce({ infrastructureType: 'advanced' });
 
   test
     .stdout()
     .stub(inquirer, 'prompt', stubResponse)
-    .command(['generate'])
-    .it('lists cloud provider', async(_, done) => {
-      done();
+    .command(['generate', projectName])
+    .it('creates a new project folder', () => {
+      expect(fs.existsSync(projectName)).to.eq(true);
```

Comment:
Should we add another test to assert the "stout" displaying a success message? (this would provide a good example for later tests when we want to assert a given question is asked, or a specific log has been displayed :) )

Reply:
Thanks for the suggestion, added in https://github.com/nimblehq/infrastructure-templates/pull/68/commits/f2e34348400404f61ba78ff8e6c5887fd78eb900